### PR TITLE
Mazda CX-5 2022 Car port

### PR DESCRIPTION
**Pull request by Jafaral**
The CX-5 2022 is the first Mazda to offer steer down to zero. It also doesn't have steer lockout.

### CODE REVIEW

**Comment by alancyau** [comment]
The steering tune below 30mph needs some loving, but above 30mph was pretty smooth.

If you set ACC below 20mph, the stock ACC still accelerate to 20mph.

File: docs/CARS.md [hunk 1]
```diff
@@ -136,6 +136,7 @@
 | Kia       | Stinger 2018                    | SCC + LKAS        | Stock            | 0mph               | 0mph         |
 | Kia       | Telluride 2020                  | SCC + LKAS        | Stock            | 0mph               | 0mph         |
 | Mazda     | CX-9 2021                       | All               | Stock            | 0mph               | 28mph        |
+| Mazda     | CX-5 2022                       | All               | Stock            | 0mph               | 0mph         |
```
Comment (adeebshihadeh) [inline docs/CARS.md:139]:
```suggestion
| Mazda     | CX-5 2022                       | All               | Stock            | 0mph               | 0mph         |
| Mazda     | CX-9 2021                       | All               | Stock            | 0mph               | 28mph        |
```

File: selfdrive/car/mazda/values.py [hunk 2]
```diff
@@ -20,6 +20,7 @@ class CAR:
   MAZDA3 = "MAZDA 3"
   MAZDA6 = "MAZDA 6"
   CX9_2021 = "MAZDA CX-9 2021"   # No Steer Lockout
+  CX5_2022 = "MAZDA CX-5 2022"   # No Steer Lockout
```
Comment (adeebshihadeh) [inline selfdrive/car/mazda/values.py:23]:
```suggestion
  CX9_2021 = "MAZDA CX-9 2021"
  CX5_2022 = "MAZDA CX-5 2022"
```

File: selfdrive/car/mazda/values.py [hunk 3]
```diff
@@ -242,10 +263,14 @@ class Buttons:
   CAR.MAZDA3: dbc_dict('mazda_2017', None),
   CAR.MAZDA6: dbc_dict('mazda_2017', None),
   CAR.CX9_2021: dbc_dict('mazda_2017', None),
+  CAR.CX5_2022: dbc_dict('mazda_2017', None),
 }
 
 # Gen 1 hardware: same CAN messages and same camera
-GEN1 = {CAR.CX5, CAR.CX9, CAR.CX9_2021, CAR.MAZDA3, CAR.MAZDA6}
+GEN1 = set([CAR.CX5, CAR.CX9, CAR.CX9_2021, CAR.MAZDA3, CAR.MAZDA6, CAR.CX5_2022])
```
Comment (adeebshihadeh) [inline selfdrive/car/mazda/values.py:270]:
```suggestion
GEN1 = {CAR.CX5, CAR.CX9, CAR.CX9_2021, CAR.MAZDA3, CAR.MAZDA6, CAR.CX5_2022}
```

File: selfdrive/car/mazda/values.py [hunk 4]
```diff
@@ -242,10 +263,14 @@ class Buttons:
...
 
 # Cars with a steering lockout
-STEER_LOCKOUT_CAR = {CAR.CX5, CAR.CX9, CAR.MAZDA3, CAR.MAZDA6}
+STEER_LOCKOUT_CAR = set([CAR.CX5, CAR.CX9, CAR.MAZDA3, CAR.MAZDA6])
+
+# Cars that don't steer below 28mph
+STEER_LIMIT_CAR = set([CAR.CX5, CAR.CX9, CAR.MAZDA3, CAR.MAZDA6, CAR.CX9_2021])
```
Comment (adeebshihadeh) [inline selfdrive/car/mazda/values.py:276]:
```suggestion
STEER_LOCKOUT_CAR = {CAR.CX5, CAR.CX9, CAR.MAZDA3, CAR.MAZDA6}

# Cars that don't steer below 28mph
STEER_LIMIT_CAR = {CAR.CX5, CAR.CX9, CAR.MAZDA3, CAR.MAZDA6, CAR.CX9_2021}
```

File: selfdrive/car/mazda/values.py [hunk 5]
```diff
@@ -242,10 +263,14 @@ class Buttons:
...
 
 # Gen 1 hardware: same CAN messages and same camera
-GEN1 = {CAR.CX5, CAR.CX9, CAR.CX9_2021, CAR.MAZDA3, CAR.MAZDA6}
+GEN1 = {CAR.CX5, CAR.CX9, CAR.CX9_2021, CAR.MAZDA3, CAR.MAZDA6, CAR.CX5_2022}
 
 # Cars with a steering lockout
 STEER_LOCKOUT_CAR = {CAR.CX5, CAR.CX9, CAR.MAZDA3, CAR.MAZDA6}
+
+# Cars that don't steer below 28mph
+STEER_LIMIT_CAR = {CAR.CX5, CAR.CX9, CAR.MAZDA3, CAR.MAZDA6, CAR.CX9_2021}
```
Comment (adeebshihadeh) [inline selfdrive/car/mazda/values.py:276]: do this in interface.py and set minSteerSpeed properly

File: selfdrive/car/mazda/interface.py [hunk 6]
```diff
@@ -22,14 +22,14 @@ def get_params(candidate, fingerprint=gen_empty_fingerprint(), car_fw=None):
     ret.safetyConfigs = [get_safety_config(car.CarParams.SafetyModel.mazda)]
     ret.radarOffCan = True
 
-    ret.dashcamOnly = candidate not in (CAR.CX9_2021,)
+    ret.dashcamOnly = candidate not in (CAR.CX5_2022, CAR.CX9_2021)
```
Comment (adeebshihadeh) [inline selfdrive/car/mazda/interface.py:25]: is it time to blacklist the dashcam only ones instead? :)

File: selfdrive/car/mazda/carstate.py [hunk 7]
```diff
@@ -65,12 +65,15 @@ def update(self, cp, cp_cam):
...
 
-    # LKAS is enabled at 52kph going up and disabled at 45kph going down
-    # wait for LKAS_BLOCK signal to clear when going up since it lags behind the speed sometimes
-    if speed_kph > LKAS_LIMITS.ENABLE_SPEED and not lkas_blocked:
-      self.lkas_allowed_speed = True
-    elif speed_kph < LKAS_LIMITS.DISABLE_SPEED:
-      self.lkas_allowed_speed = False
+    carFP = self.CP.carFingerprint
+
+    if carFP not in (CAR.CX5_2022):
```
Comment (adeebshihadeh) [inline selfdrive/car/mazda/carstate.py:70]: set carParams.minSteerSpeed in interface, then check that here

File: selfdrive/car/mazda/interface.py (hunk 6 above)
Comment (Jafaral) [reply selfdrive/car/mazda/interface.py:25]: Not yet! 4 vs 2.

File: selfdrive/car/mazda/carstate.py (hunk 7 above)
Comment (Jafaral) [reply selfdrive/car/mazda/carstate.py:70]: Set minSteetSpeed appropriately per supported car you mean?

Comment (adeebshihadeh) [reply selfdrive/car/mazda/carstate.py:70]: yeah

Comment (Jafaral) [reply selfdrive/car/mazda/carstate.py:70]: Can we leave this as-is for now ?  I need to do some clean up to prepare support for newer gen Mazdas

Comment (adeebshihadeh) [reply selfdrive/car/mazda/carstate.py:70]: I'd like to do this one properly. I can do it if you want

Comment (Jafaral) [reply selfdrive/car/mazda/carstate.py:70]: Sure, otherwise, I can do it late tonight. Thanks @adeebshihadeh

Comment (Jafaral) [reply selfdrive/car/mazda/carstate.py:70]: Gave you head start @adeebshihadeh!

File: selfdrive/car/mazda/interface.py [hunk 8]
```diff
@@ -58,8 +58,8 @@ def get_params(candidate, fingerprint=gen_empty_fingerprint(), car_fw=None):
       ret.lateralTuning.pid.kpV, ret.lateralTuning.pid.kiV = [[0.19], [0.019]]
       ret.lateralTuning.pid.kf = 0.00006
 
-    # No steer below disable speed
-    ret.minSteerSpeed = LKAS_LIMITS.DISABLE_SPEED * CV.KPH_TO_MS
+    if candidate not in (CAR.CX5_2022):
```
Comment (adeebshihadeh) [inline selfdrive/car/mazda/interface.py:61]:
```suggestion
    if candidate not in (CAR.CX5_2022, ):
```

**Pull request merged**